### PR TITLE
Closes #40, Closes #38 Allow simultaneously maintaining releases for 5.x and 2.x versions of Arizona Bootstrap docs via CI infrastructure.

### LINKED ISSUE #38: Update workflow to push docs to version buckets instead of syncing 2.0 folder
## Conditions of satisfaction
- [x] Update https://github.com/az-digital/digital.arizona.edu/blob/main/.github/workflows/release.yml to push docs to versioned folders.

### LINKED ISSUE #40: Allow simultaneously maintaining releases for 5.x and 2.x versions of Arizona Bootstrap docs via CI infrastructure.
## Motivation

In order to simultaneously maintain two separate versions of Arizona Bootstrap Docs within this repository, we need a way to choose which build script is run.

### PR DESCRIPTION
**Pull request by trackleft**
Blocked by 

- [x] Update README to describe the purpose of this repository and how the actions work.
- [x] Decision on what to do with this page https://digital.arizona.edu/arizona-bootstrap/ which does not exist in the 2.0 folder

Could possibly add the page here: https://github.com/az-digital/digital.arizona.edu/pull/41/files#diff-3b8dc37ada427fbf61674bb7a58f15eefd024fedf1d4de79410c2f110f455c53R87
Until we are ready for a 5.x release.

I this PR I add 2 new workflow files:
- release-branch-trigger.yml
- deploy-docs-for-v2-release.yml

Renamed 
- release.yml -> deploy-docs-for-v5-release.yml

### What do these changes accomplish
Allow simultaneously maintaining releases for 5.x and 2.x versions of Arizona Bootstrap docs via CI infrastructure.

#### release-branch-trigger.yml
Repository dispatches can only be run on the main branch.
Repository dispatch type of workflow that accepts the branch name, version and sha from the arizona bootstrap repository
Conditionally passes repo dispatch variables and runs version of doc building workflow depending on the branch that the arizona-bootstrap release was on.

if `2.x` runs deploy-docs-for-v2-release.yml
if `main` runs deploy-docs-for-v5-release.yml

#### deploy-docs-for-v2-release.yml
Should not be run manually
Is triggered by release-branch-trigger.yml
Only pushes updates into the `docs/arizona-bootstrap/docs/2.0` folder

#### deploy-docs-for-v5-release.yml
Should not be run manually
Is triggered by release-branch-trigger.yml
Pushes updates into the docs/arizona-bootstrap folder but excludes the `docs/arizona-bootstrap/docs/2.0` folder

### Alternatives considered
One repository_dispatch workflow action that switches behavior depending on the branch that triggers the workflow

## How to test
Create 2 releases on https://github.com/trackleft/arizona-bootstrap/actions/workflows/release.yml
From the `main` branch, create a 5.x release
From the `2.x` branch, create a 2.x release

### CODE REVIEW

File: .github/workflows/v2-release-docs.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+      - name: Build and deploy documentation site artifact
+        run: |
+          sudo touch "${AZ_BOOTSTRAP_CLONE_DIR}/config.yml"
+          sudo find "$AZ_BOOTSTRAP_CLONE_DIR" -path "${AZ_BOOTSTRAP_CLONE_DIR}/.git" -prune -o -exec chown 1000:1000 {} \;
+          sudo chown 1000:1000 "$AZ_BOOTSTRAP_CLONE_DIR"
+          docker run --rm -e "AZ_SITE_BASE_URL=${AZ_REVIEW_BASEURL}" -e "AZ_SITE_HOST=${AZ_SITE_HOST}" -v "$AZ_BOOTSTRAP_CLONE_FULL_PATH":"/${AZ_BOOTSTRAP_SOURCE_DIR}" "$AZ_EPHEMERAL_IMAGE" expose-review-site
+          rsync --recursive --delete "${AZ_BOOTSTRAP_CLONE_DIR}/_site/" "docs${AZ_REVIEW_BASEURL}"
+          git config --global user.email "${GITHUB_ACTOR}@users.noreply.github.com"
+          git config --global user.name "${GITHUB_ACTOR}"
+          git add docs/2.0
```
Comment (trackleft) [inline .github/workflows/v2-release-docs.yml:86]:
```suggestion
          git add docs/arizona-bootstrap/docs/2.0
```

Comment (trackleft) [inline .github/workflows/v2-release-docs.yml:86]:
```suggestion
          git add docs/arizona-bootstrap/docs/2.0
```

File: .github/workflows/v2-release-docs.yml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+          git commit -m "Arizona Bootstrap updates for ${{ github.event.inputs.version }}"
+          git push
```
Comment (trackleft) [inline .github/workflows/v2-release-docs.yml:88]:
```suggestion
          git push
      
```

File: .github/workflows/release-branch-trigger.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+name: Trigger docs deploy on specific branch release
```
Comment (trackleft) [inline .github/workflows/release-branch-trigger.yml:1]: Compare with working version on fork https://github.com/az-digital/digital.arizona.edu/pull/39/files#diff-78aa009faa8c3019d2decd00ced30a565bbf48804e5f2198e43fc41cb3e050e1

File: .github/workflows/deploy-docs-for-v5-release.yml [hunk 4]
```diff
@@ -1,19 +1,26 @@
-name: Create Release
```
Comment (trackleft) [inline .github/workflows/deploy-docs-for-v5-release.yml:1]: Compare with working version on fork https://github.com/az-digital/digital.arizona.edu/pull/39/files#diff-cb829693f3c0464fd1780c6cb3f899c1a5101f506f09fd1e30c5cbc89e71d489

File: .github/workflows/deploy-docs-for-v2-release.yml [hunk 5]
```diff
@@ -0,0 +1,89 @@
+name: Build and deploy documentation for 2.x
```
Comment (trackleft) [inline .github/workflows/deploy-docs-for-v2-release.yml:1]: Compare with working version on fork https://github.com/az-digital/digital.arizona.edu/pull/39/files#diff-cb829693f3c0464fd1780c6cb3f899c1a5101f506f09fd1e30c5cbc89e71d489

File: .github/workflows/deploy-docs-for-v2-release.yml [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+      - name: Build and deploy documentation site artifact
+        run: |
+          sudo touch "${AZ_BOOTSTRAP_CLONE_DIR}/config.yml"
+          sudo find "$AZ_BOOTSTRAP_CLONE_DIR" -path "${AZ_BOOTSTRAP_CLONE_DIR}/.git" -prune -o -exec chown 1000:1000 {} \;
+          sudo chown 1000:1000 "$AZ_BOOTSTRAP_CLONE_DIR"
+          docker run --rm -e "AZ_SITE_BASE_URL=${AZ_REVIEW_BASEURL}" -e "AZ_SITE_HOST=${AZ_SITE_HOST}" -v "$AZ_BOOTSTRAP_CLONE_FULL_PATH":"/${AZ_BOOTSTRAP_SOURCE_DIR}" "$AZ_EPHEMERAL_IMAGE" expose-review-site
+          rsync --recursive --delete "${AZ_BOOTSTRAP_CLONE_DIR}/_site/" "docs${AZ_REVIEW_BASEURL}"
+          git config --global user.email "${GITHUB_ACTOR}@users.noreply.github.com"
+          git config --global user.name "${GITHUB_ACTOR}"
+          git add docs/arizona-bootstrap/docs/2.0
```
Comment (trackleft) [inline .github/workflows/deploy-docs-for-v2-release.yml:87]: This line is what makes it so that releases only update pages within the docs/2.0 folder

Comment (trackleft) [reply .github/workflows/deploy-docs-for-v2-release.yml:87]:
```suggestion
          git add docs/arizona-bootstrap
```

**Review by mmunro-ltrr — APPROVED** [review]

**Review by joeparsons — APPROVED** [review]

**Pull request merged**